kse/src/main/java/org/kse/gui/actions/AuthorityCertificatesVerifyAction.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2004 - 2013 Wayne Grant
+ *           2013 - 2025 Kai Kramer
+ *
+ * This file is part of KeyStore Explorer.
+ *
+ * KeyStore Explorer is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * KeyStore Explorer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with KeyStore Explorer.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.kse.gui.actions;
+
+import java.security.KeyStore;
+import java.security.Security;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.jcajce.JcaCertStore;
+import org.bouncycastle.util.Store;
+import org.kse.AuthorityCertificates;
+import org.kse.crypto.CryptoException;
+import org.kse.crypto.SecurityProvider;
+import org.kse.crypto.x509.X509CertUtil;
+import org.kse.gui.KseFrame;
+import org.kse.utilities.history.KeyStoreHistory;
+import org.kse.utilities.history.KeyStoreState;
+
+/**
+ * Abstract base class for actions that utilize authority certificates for establishing
+ * trust to third-party certificate authorities (e.g., time stamping server certificates).
+ */
+public abstract class AuthorityCertificatesVerifyAction extends AuthorityCertificatesAction {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Construct action.
+     *
+     * @param kseFrame KeyStore Explorer frame
+     */
+    public AuthorityCertificatesVerifyAction(KseFrame kseFrame) {
+        super(kseFrame);
+    }
+
+    /**
+     * Get the complete set of trusted certificates for establishing trust during signature
+     * verification.
+     *
+     * @return Set<X509Certificate>
+     * @throws CryptoException If there was a problem getting the trusted certificates.
+     */
+    protected Set<X509Certificate> getTrustedCertificates() throws CryptoException {
+        KeyStoreHistory history = kseFrame.getActiveKeyStoreHistory();
+
+        KeyStoreState currentState = history.getCurrentState();
+        KeyStore keyStore = currentState.getKeyStore();
+
+        KeyStore caCertificates = getCaCertificates();
+        KeyStore windowsTrustedRootCertificates = getWindowsTrustedRootCertificates();
+
+        // Perform cert lookup against current KeyStore
+        Set<X509Certificate> trustedCerts = new HashSet<>();
+        trustedCerts.addAll(extractCertificates(keyStore));
+
+        if (caCertificates != null)
+        {
```

Comment:
Please try to keep the code formatting consistent. There are several other occurrences of this in this PR.